_posts/2022-08-15-python-conda-environments-create.md

```diff
@@ -0,0 +1,377 @@
+
+I didn't want to students to spend weeks trying to get an environment working.
+That is how you lose people.
+
+> If I can't install software, how can I code?
+> -- Says every student ever who deals with imposter syndrome
+
+
+
+So what was my next step?
+
+## I (we) built the Earth Analytics Python Environment (& setup testing)
+
+I knew that having an easier-to-install / almost fail proof Python environment 
+would make teaching easier.
+And better yet, if I could test that environment on Linux, Mac and Windows 
+(students use all sorts of computers and OS' - you can't assume only one) then
+a lot of my issues would be solved. 
+I would know before they installed the environment where the pain points were. I could troubleshoot before class rather than during class. 
+
+This was huge!
+
+### Enter the Earth Analytics Python Environment
+
+So this is what I built:
+
+<i class="fas fa-hands-helping"></i> Thanks to the help of several people 
+including a great colleague 
+[Max Joseph](https://www.github.com/mbjoseph/). Max was really the person who
+early on introduced me to the concept of testing. At the time I didn't fully
+understand just how important it was. But I got there quickly!
+{: .notice }
+
+1. [A vetted Python environment - this links to the original repo on EarthLab's organization. Below I link to a fork in case that repo changes over time](https://www.github.com/earthlab/earth-analytics-python-env/): This environment has all of the Python tools needed to complete my earth-analytics-python class. Tools for remote sensing and other spatial data. Tools for natural language processing, etc. And it can be installed quickly and easily using a recipe file (also known as an environment file - more on that below).
+2. The environment has a [GitHub Actions workflow that tested it on Windows, Mac and Linux.](https://github.com/lwasser/earth-analytics-python-env/tree/main/.github/workflows). I'll talk about that part more in a second, followup blog. 
+
+*The original workflow was built using Travis CI and Appveyor - yikes!*
+
+<i class="fas fa-hands-helping"></i> Thanks to [Filipe's help](https://www.github.com/ocefpaf), in a separate 
+environment that I created for an automated build, we used Conda Lock. We did 
+this so we could pin down exactly what versions of all tools and dependencies 
+were being installed once we have a working environment. Boy this made 
+troubleshooting easier! More on that in another blog.
+{: .notice}
+
+## What do all of the pieces in this earth-analytics-python repository do?
+
+Ok so let's break down what is in this [`earth-analytics-python` GitHub repo](https://www.github.com/lwasser/earth-analytics-python-env):
+
+### First you have the conda Python environment file. What's that? 
+
+You can think of a `Python` environment file, as a recipe for
+what to install and where. By using this environment file, 
+we are making a decision now to use the `conda` package manager 
+to install packages vs `pip`. 
```

Comment:
```suggestion
You can think of a `Python` environment file, as a recipe for
what to install and where. The environment file, 
tells the installer how to install each tool.  It can also be used to specify specific tool versions. And you can tell it what package manager and channel to use:
* `pip` vs `conda`,
* `conda-forge` vs `defaults`,
etc. 
```